common/lib/capa/capa/responsetypes.py

```diff
@@ -1515,7 +1515,9 @@ def get_score(self, student_answers):
         log.debug('%s: student_answers=%s', unicode(self), student_answers)
 
         # ordered list of answer id's
-        idset = sorted(self.answer_ids)
+        # sort the responses on the basis of position no which is at last place in id and
```

Comment:
```
sort the responses on the bases of the problem's position number, which can be found in the last place in the problem id. Then convert this number into an int, so that we sort on ints instead of strings
```